dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java

```diff
@@ -166,19 +160,27 @@ public final Response deleteApiToken(@Context final HttpServletRequest request,
     @NoCache
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public final Response issueApiToken(@Context final HttpServletRequest request,
-                                   @Context final HttpServletResponse response,
-                                   final ApiTokenForm formData) {
+    public final Response issueApiToken(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            final ApiTokenForm formData) throws DotDataException, DotSecurityException {
         
         final InitDataObject initDataObject = this.webResource.init(null, true, request, true, null);
         final User requestingUser = initDataObject.getUser();
-        final User forUser = Try.of(()->APILocator.getUserAPI().loadUserById(formData.userId,requestingUser, false )).getOrNull();
-        
-        if(forUser == null) {
+        final User forUser = formData.userId != null ? getUserById(formData, requestingUser) : requestingUser;
+
+        if (requestingUser != forUser && !requestingUser.isAdmin()) {
+            throw new DotDataException("Just Admin user can request a Token for another user");
+        }
 
+        if(forUser == null) {
```

Comment:
Shouldn't this be the first evaluation?

Reply:
done https://github.com/dotCMS/core/pull/19698/commits/95a2812fa9c6ac8f3ec611d7223a4716794ad726